src/solve/execute-route.ts

```diff
@@ -0,0 +1,308 @@
+import {
+  enableChains,
+  getAddressForChain,
+  getChainByID,
+  getOfflineSigner,
+  getOfflineSignerOnlyAmino,
+  getSigningCosmWasmClientForChainID,
+  getSigningStargateClientForChainID,
+  getStargateClientForChainID,
+  isLedger,
+  signAmino,
+  signAndBroadcastEvmos,
+  signAndBroadcastInjective,
+} from "@/utils/utils";
+import { EncodeObject, OfflineSigner, coin } from "@cosmjs/proto-signing";
+import { GasPrice, DeliverTxResponse } from "@cosmjs/stargate";
+import { MsgTransfer as MsgTransferInjective } from "@injectivelabs/sdk-ts";
+import { WalletClient } from "@cosmos-kit/core";
+import { SkipClient, MsgsRequest, RouteResponse } from "./client";
+import Long from "long";
+import { OfflineAminoSigner } from "@keplr-wallet/types";
+import { TxRaw } from "cosmjs-types/cosmos/tx/v1beta1/tx";
+
+const wait = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export async function executeRoute(
+  skipClient: SkipClient,
+  walletClient: WalletClient,
+  route: RouteResponse,
+  onTxSuccess: (tx: any, index: number) => void,
+  onError: (error: any) => void
+) {
+  await enableChains(walletClient, route.chain_ids);
+
+  const userAddresses: Record<string, string> = {};
+
+  // get addresses
+  for (const chainID of route.chain_ids) {
+    const address = await getAddressForChain(walletClient, chainID);
+
+    userAddresses[chainID] = address;
+  }
+
+  const msgRequest: MsgsRequest = {
+    source_asset_denom: route.source_asset_denom,
+    source_asset_chain_id: route.source_asset_chain_id,
+    dest_asset_denom: route.dest_asset_denom,
+    dest_asset_chain_id: route.dest_asset_chain_id,
+    amount_in: route.amount_in,
+    chain_ids_to_addresses: userAddresses,
+    operations: route.operations,
+
+    estimated_amount_out: route.estimated_amount_out,
+    slippage_tolerance_percent: "5.0",
+    affiliates: [],
+  };
+
+  const msgsResponse = await skipClient.fungible.getMessages(msgRequest);
+
+  // check balances on chains where a tx is initiated
+  for (let i = 0; i < msgsResponse.msgs.length; i++) {
+    const multiHopMsg = msgsResponse.msgs[i];
+
+    const chain = getChainByID(multiHopMsg.chain_id);
+
+    const client = await getStargateClientForChainID(multiHopMsg.chain_id);
+
+    const feeInfo = chain.fees?.fee_tokens[0];
+
+    if (!feeInfo) {
+      throw new Error("No fee info found");
+    }
+
+    let gasNeeded = 200000;
+    if (
+      route.does_swap &&
+      route.swap_venue?.chain_id === multiHopMsg.chain_id
+    ) {
+      gasNeeded = 1000000;
```

Comment:
move to a helper function and move the hard code values to constants that can be discovered more easily and changed in one place